presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java

```diff
@@ -1341,6 +1365,33 @@ private static OGCGeometry geomFromBinary(Slice input)
         return geometry;
     }
 
+    private static OGCGeometry geomFromHadoopShape(Slice input)
+    {
+        requireNonNull(input, "input is null");
+
+        byte[] bytes = input.getBytes();
+
+        try {
+            OGCGeometry geometry = OGCGeometry.fromEsriShape(getShapeByteBuffer(bytes));
+            String wkt = GeometryEngine.geometryToWkt(geometry.getEsriGeometry(), getWktExportFlags(bytes));
+            return OGCGeometry.fromText(wkt);
+        }
+        catch (IndexOutOfBoundsException e) {
```

Comment:
Instead of catching IOOB here, it would be cleaner to explicitly check the value against the array size. Then we can throw a message like
```java
"Invalid Hadoop shape type: " + type
```
Also, we should explicitly validate the size of the input against the minimum length and throw a message like
> Hadoop shape input is too short

Reply:
I refactored a little bit to be specific when possible, but `OGCGeometry.fromEsriShape` can fail with an IndexOutOfBoundsException or UnsupportedOperationException if the ByteBuffer is not a valid esri shape. Additionally, `GeometryEngine.geometryToWkt` will throw an `IllegalArgumentException` if the shape given can not be serialized into the type given.

Please let me know if I can refactor this any further.